### PR TITLE
Dashboard unavailability, network fix for Flannel and Canal #1394

### PR DESCRIPTION
**Pull request by rafzei**

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/calico.yml [hunk 1]
```diff
@@ -34,4 +36,4 @@
 - name: Include Calico deployment tasks
   include_tasks: deployments/deploy-template.yml
   vars:
-    file_name: calico.yml.j2
+    file_name: calico.yml.j2
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/calico.yml:39]: please add newline

File: core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml [hunk 2]
```diff
@@ -1,4 +1,39 @@
...
+- name: Reload NetworkManager service
+  when: copy_flannel_network_manager_conf.changed
+  block:
+    - name: Populate service facts
+      service_facts:
+
+    - name: Reload if NetworkManager service is running
+      systemd:
+        name: NetworkManager
+        state: reloaded
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml:34]: Could you comment on why we need to reload NM? It's not obvious from the code itself (at least for me), and I believe this isn't always a safe operation.

File: core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/flannel.yml [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+- name: Reload NetworkManager service
+  when: copy_flannel_network_manager_conf.changed
+  block:
+    - name: Populate service facts
+      service_facts:
+
+    - name: Reload if NetworkManager service is running
+      systemd:
+        name: NetworkManager
+        state: reloaded
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/flannel.yml:34]: Could you comment on why we need to reload NM? It's not obvious from the code itself (at least for me), and I believe this isn't always a safe operation.

File: core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml [hunk 4]
```diff
@@ -1,4 +1,39 @@
...
+    - name: Check if /etc/NetworkManager/conf.d exists
+      stat:
+        path: /etc/NetworkManager/conf.d
+        get_attributes: no
+      register: network_manager_conf_d_stat
+
+    # Remove flannel.1 interface from NetworkManager managed devices
+    - name: Copy flannel.conf to /etc/NetworkManager/conf.d
+      template:
+        src: network-manager.conf.j2
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml:14]: can the file be named `flannel.conf.j2` if it's `flannel.conf`?

File: core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/flannel.yml [hunk 5]
```diff
@@ -1,5 +1,40 @@
...
+    - name: Check if /etc/NetworkManager/conf.d exists
+      stat:
+        path: /etc/NetworkManager/conf.d
+        get_attributes: no
+      register: network_manager_conf_d_stat
+
+    # Remove flannel.1 interface from NetworkManager managed devices
+    - name: Copy flannel.conf to /etc/NetworkManager/conf.d
+      template:
+        src: network-manager.conf.j2
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/flannel.yml:14]: same here

File: core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml (hunk 4 above)
Comment (rafzei) [reply core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml:15]: @mkyc network-manager.conf.j2 is a base template for NetworkManager configuration all CNI plugins. Keeping a separated template for each CNI is redundant.

**Pull request merged**